src/methods/properties.js

```diff
@@ -1005,7 +1007,20 @@ export function OrdinaryGetOwnProperty(realm: Realm, O: ObjectValue, P: Property
   // 2. If O does not have an own property with key P, return undefined.
   let existingBinding = InternalGetPropertiesMap(O, P).get(InternalGetPropertiesKey(P));
   if (!existingBinding) {
-    if (O.isPartialObject() && !O.isSimpleObject()) {
+    if (O.isPartialObject()) {
+      invariant(realm.useAbstractInterpretation); // __makePartial will already have thrown an error if not
+      if (O.isSimpleObject()) {
+        if (P instanceof StringValue) P = P.value;
+        if (typeof P === "string") {
+          // In this case it is safe to defer the property access to runtime
+          invariant(realm.generator);
+          let pname = realm.generator.getAsPropertyNameExpression(P);
+          let absVal = realm.deriveAbstract(TypesDomain.topVal, ValuesDomain.topVal, [O], ([node]) =>
+            t.memberExpression(node, pname, !t.isIdentifier(pname))
```

Comment:
Ah, I incorrectly looked at the code to be "t.Identifier()" instead of "t.isIdentifier()".